src/meta/src/controller/catalog/drop_op.rs

```diff
@@ -15,6 +15,64 @@
 use super::*;
 
 impl CatalogController {
+    // drop table associated source is a special case of drop relation, which just remove the source object and associated state table, keeping the streaming job and fragments.
+    pub async fn drop_table_associated_source(
```

Comment:
@yezizp2012 can you help take a look? seems the func accidentally dropped the table catalog along with the state table's. 
Let's say table_id = 6, source_id = 7, source_state_table = 8, the release context controls dropping 7 and 8, but 6 seems to be dropped in cascade.

Reply:
Need to update `optional_associated_source_id` of table 6 to `Null`.